src/ota_metadata/file_table/_table.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2022 TIER IV, INC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import annotations
+
+import os
+import shutil
+import stat
+from pathlib import Path
+from typing import ClassVar, Literal, Optional
+
+from pydantic import BaseModel, SkipValidation
+from simple_sqlite3_orm import ConstrainRepr, TableSpec, TypeAffinityRepr
+from typing_extensions import Annotated
+
+from ota_metadata.file_table._types import EntryAttrsType
+from otaclient_common.typing import StrOrPath
+
+CANONICAL_ROOT = "/"
+
+PrepareMethod = Literal["move", "hardlink", "copy"]
```

Comment:
If this definition is used only in `FileTableRegularFiles` class, we can narrow the scope.

Reply:
Good point! Let's move it under the FileTableRegularFiles class.

Reply:
Have some try, and pydantic will complains about it if we directly put the `PrepareMethod` type alias under `FileTableRegularFiles`. 
If we type hint the `PrepareMethod` type alias as `ClassVar[TypeAlias]`, now the pylance will complain about the typing :skull: .
So finally I just directly using `Literal["move", "hardlink", "copy"]` on typing the `prepare_method` arg without defining the global `PrepareMethod` type alias.

Reply:
Updated in [4fefa64](https://github.com/tier4/ota-client/pull/460/commits/4fefa641440c0d20a626736e03272df735508341)